metricbeat/schema/schema_test.go

```diff
@@ -37,6 +37,21 @@ func TestSchema(t *testing.T) {
 	})
 }
 
+func TestHasKey(t *testing.T) {
+	schema := Schema{
+		"test": Conv{Key: "test", Func: nop},
+		"test_obj": Object{
+			"test_a": Conv{Key: "testA", Func: nop},
+			"test_b": Conv{Key: "testB", Func: nop},
+		},
+	}
+
+	assert.True(t, schema.HasKey("test"))
+	assert.True(t, schema.HasKey("test_b"))
+	assert.True(t, schema.HasKey("test_a"))
+	assert.False(t, schema.HasKey("nooo"))
+}
```

Comment:
I don't think `HasKey` should return true for both the "source" keys and the "output" keys. Here's the behavior I expected which returns true only for the "source" keys.

```
func TestHasKey(t *testing.T) {
    schema := Schema{
        "test": Conv{Key: "Test", Func: nop},
        "test_obj": Object{
            "test_a": Conv{Key: "TestA", Func: nop},
            "test_b": Conv{Key: "TestB", Func: nop},
        },  
    }   

    assert.True(t, schema.HasKey("Test"))
    assert.True(t, schema.HasKey("TestA"))
    assert.True(t, schema.HasKey("TestB"))
    assert.False(t, schema.HasKey("test"))
    assert.False(t, schema.HasKey("test_obj"))
    assert.False(t, schema.HasKey("test_a"))
    assert.False(t, schema.HasKey("test_b"))
    assert.False(t, schema.HasKey("other"))
}
```

which would mean that `hasKey()` would be:

```
func hasKey(key string, mappers map[string]Mapper) bool {
	for _, mapper := range mappers {
		if mapper.HasKey(key) {
			return true
		}
	}
	return false
}
```